### DISCUSSION TITLE
Compare date columns

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#10310] [Questions] [answered]
Trying to figure out the correct way of comparing three date columns:

`.or('DATE1.is.null,DATE1.lt.DATE2,DATE1.lt.DATE3')`

Throws "invalid input syntax for type timestamp". Is it the syntax above or the use of "lt" that is wrong. Or both :/

**Comment by GaryAustin1** [comment, marked as answer]
IF DATE1,DATE2,DATE3 are columns in your database that is not going to work.
The API can only have one column and the other value must be a number/string/date value.
You can not compare two columns with the API.
You can do want you want with an rpc function and SQL.

Also you should not use capital letters for Postgres names of tables/columns/functions unless you are forced to.  You must always remember to enclose them in double quotes in SQL code (and you will forget).  Like "DATE1" or "camelCase".

**Reply by ghost** [reply]
Thanks for clarifying, and for the suggestions.
